parser/src/token.rs

```diff
@@ -516,6 +548,47 @@ impl<'input> Iterator for Tokenizer<'input> {
     }
 }
 
+/// Converts partial hex literal (i.e. part after `0x` or `-0x`) to 64 bit signed integer.
+///
+/// This is basically a copy and adaptation of `std::num::from_str_radix`.
+fn i64_from_hex(hex: &str, is_positive: bool) -> Result<i64, Error> {
+    let radix = 16;
+    let digits = hex.as_bytes();
+    let mut result = 0i64;
+    if is_positive {
+        for &c in digits {
+            let x = match (c as char).to_digit(radix) {
```

Comment:
The string is already validated to be all hex digits at this point so just do `expect` here which also makes it possible to remove the `HexLiteralInvalid` case.